dora/core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -517,8 +522,24 @@ public Type getType() {
     mActiveSyncMetadataExecutor.allowCoreThreadTimeOut(true);
     FileSystemContext schedulerFsContext = FileSystemContext.create();
     JournaledJobMetaStore jobMetaStore = new JournaledJobMetaStore(this);
-    mScheduler = new Scheduler(schedulerFsContext,
-        new DefaultWorkerProvider(this, schedulerFsContext), jobMetaStore);
+    WorkerProvider workerProvider;
+    switch (Configuration.getEnum(PropertyKey.WORKER_MEMBERSHIP_MANAGER_TYPE,
+        MembershipType.class)) {
+      case STATIC:
+        workerProvider = new MembershipManagerWorkerProvider(
+            StaticMembershipManager.create(Configuration.global()), schedulerFsContext);
+        break;
+      case ETCD:
+        workerProvider = new MembershipManagerWorkerProvider(
+            EtcdMembershipManager.create(Configuration.global()), schedulerFsContext);
```

Comment:
u could collapse these two and do 
new MembershipManagerWorkerProvider(
   MembershipManager.create()...
)

Reply:
done